arduino/src/arduino.cpp

```diff
@@ -0,0 +1,444 @@
+}
+/**
+ * @brief will get a data dump of the arduino to a file 
+ * the func will send a commend to the arduino to start sending the data
+ * and then will recive READLENGTH of bytes. the func will write them to the file
+ * that is privided.
+ * the problem right now is that it has only 1000 bytes of data.
+ * @param Wsensor which sensor
+ * @param place the place in the array of the files
+ * @return int the status of result
+ */
+Arduino::ERRORARDUINO Arduino::getSensorDataB(int Wsensor, int place)
+// TODO maby the sensor param is stupid.
+{
+    if (!this->isOpen)
+        return Arduino::ERRORARDUINO::ERROR_WITH_SERIALPORT; // TODO it will check twice once inside this func and once inside the serialCommend.
+    std::string commendToArduino = std::to_string(Arduino::HEADERS::ASK_FOR_STATUS_PER_SENSOR);
+    commendToArduino.append(std::to_string(Wsensor)); // the protocol works like http (more on that in protocol doc)
+    if (serialCommend(commendToArduino) != Arduino::ERRORARDUINO::ERROR_WITH_SERIALPORT)
+        ;
+    {
+        return Arduino::ERRORARDUINO::ERROR_WITH_SERIALPORT;
+    }
+    char data[DATALEN] = {0};                                 // TODO   check if 1000 is enough.
+    this->serial.readString(data, '\n', READLENGTH, TIMEOUT); // TODO the timout can be for the entire send,
+    //that can cause the readString func to return without all the data
+    writeFromBufferToFile(data, place);
+    return Arduino::ERRORARDUINO::SUCCSESS;
+}
+/**
+ * @brief dumps data to file +timestamp
+ * @param data 
+ * @param place from the file array
+ * @return int 
+ */
+int Arduino::writeFromBufferToFile(std::string data, int place)
+{
+    std::ofstream dataLog;
+    // const char* name = this->files[place].c_str();
+    // dataLog.open(this->files[place].c_str(), std::ios::app);
+    //TODO   i just couldnt open it with the relative path
+    dataLog.open("/home/pi/garden-chicken_coop/arduino/dataFromArduino/sensor4.log", std::ios::app);
+    if (!dataLog)
+    {
+        std::cerr << "Could not open the file!" << std::endl;
+        std::cout << system("ls") << std::endl;
+    }
+    // TODO the time ↓ cannot be trusted
+    dataLog << std::time(0) << std::endl;
+    dataLog << data << std::endl;
+    dataLog.close();
+    return SUCCESS;
+}
+int Arduino::getCheckSumFromMessage(std::string massage)
+{
+    std::size_t foundExamation = massage.find('!');
+    std::size_t foundn = massage.find('\n');
+    if (foundn == std::string::npos)
+    {
+        // the end line wasnt recicved well
+    }
+    if (foundExamation == std::string::npos)
+    {
+        // the ! was not found
+    }
+    std::string checkString = massage.substr(foundExamation + 1, foundn - (foundExamation + 1));
+    // TOD-1 meby put all of this to a function
+    //~foundN - foundExamation~ the distance between the satrt and the end.
+    // this way it will give a positive result
+    std::string::size_type sz; // alias of size_t
```

Comment:
if i what to know what is this then i will check this you dont need to tell me in the comment